### DISCUSSION TITLE
Google Calendar Intergration

### DISCUSSION THREAD

**Discussion by matthewbirtch** [mattermost/community#42] [Practical Ideas]
This is a big missing piece IMO as a workplace collaboration tool. I believe work is being done on this, but I'd love to prioritize this. 

- have an RHS view of 'my day' in MM
- ability to get notifications of new events I'm invited to and accept/reject them all in MM
- Some ability to create new events (might be a lot harder) from within MM

**Comment by larkox** [comment]
I think @mickmister had something working with this.

**Reply by mickmister** [reply]
Yep I have a POC of augmenting the mscalendar plugin to support Google calendar here https://github.com/mattermost/mattermost-plugin-mscalendar/pull/210. I'd like to continue building out the missing pieces to make it feature complete with the foundation in the mscalendar plugin. Then we can continue down the path of implementing more features.

**Comment by esethna** [comment]
Also want to get notifications when events are upcoming or starting with relevant info, like zoom link and description with resources

**Comment by Willyfrog** [comment]
I wonder if it would be possible to tie the agenda plugin to the calendar one, so you can not only add events to a recurring event but to any

**Reply by larkox** [reply]
In theory is possible. But we would have to dig deeper in what would be the UX.

**Comment by chetanyakan** [comment]
I would also like to be able to see my calendar in an iframe or something, so I can see my day at a glance without leaving mattermost.

**Reply by esethna** [reply]
Yep, I think that's what being proposed to use the RHS for that

**Comment by hahmadia** [comment]
I really am hoping this wins!

Google calendar is definitely the application I tend to switch over from Mattermost the most. This is the integration I've been wanting ever since joining Mattermost.

Having this idea of Mattermost suite where everything can be done within Mattermost without switching context and considering how many companies use Google Suite for managing their things, I feel this is a must have.
 
 Having spoken to @mickmister , I know he's got a big chunk of this as POC. I'm hoping this wins so we can get his POC PR rolling forward.

**Comment by zefhemel** [comment]
Another aspect of this integration that I've seen with "certain other messengers" is to auto update a person's status icon based on whether they're in a meeting or not (e.g. a calendar icon when they're in a meeting, or switch on "do not disturb" automatically).

**Reply by mickmister** [reply]
@zefhemel This is supported by the [Microsoft Calendar plugin](https://github.com/mattermost/mattermost-plugin-mscalendar), which is being used as the baseline for the Google Calendar plugin, so this is basically already implemented.

**Comment by zefhemel** [comment]
And regarding the "create a new event" it would be cool to actually leverage the fact this is a messenger, so that if you try to create a new event in a direct message channel, it would allow you to share different options to schedule a meeting (based on availability), and in a larger channel have some sort of date/time slot voting mechanism like [Doodle](https://doodle.com/en/).
